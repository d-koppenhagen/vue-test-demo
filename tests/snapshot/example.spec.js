import { shallowMount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'

describe('HelloWorld.vue', () => {
  it('renders the component completely', () => {
    const msg = 'new message';
    const wrapper = shallowMount(HelloWorld, {
      props: { msg }
    });
    expect(wrapper).toMatchSnapshot();
  })
})
