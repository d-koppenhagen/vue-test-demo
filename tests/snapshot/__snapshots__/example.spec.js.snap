// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue renders the component completely 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {},
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="hello"
      >
        <h1>
          new message
        </h1>
        <p>
           For a guide and recipes on how to configure / customize this project,
          <br />
           check out the 
          <a
            href="https://cli.vuejs.org"
            rel="noopener"
            target="_blank"
          >
            vue-cli documentation
          </a>
          . 
        </p>
        <h3>
          Installed CLI Plugins
        </h3>
        <ul>
          <li>
            <a
              href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-babel"
              rel="noopener"
              target="_blank"
            >
              babel
            </a>
          </li>
          <li>
            <a
              href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint"
              rel="noopener"
              target="_blank"
            >
              eslint
            </a>
          </li>
        </ul>
        <h3>
          Essential Links
        </h3>
        <ul>
          <li>
            <a
              href="https://vuejs.org"
              rel="noopener"
              target="_blank"
            >
              Core Docs
            </a>
          </li>
          <li>
            <a
              href="https://forum.vuejs.org"
              rel="noopener"
              target="_blank"
            >
              Forum
            </a>
          </li>
          <li>
            <a
              href="https://chat.vuejs.org"
              rel="noopener"
              target="_blank"
            >
              Community Chat
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/vuejs"
              rel="noopener"
              target="_blank"
            >
              Twitter
            </a>
          </li>
          <li>
            <a
              href="https://news.vuejs.org"
              rel="noopener"
              target="_blank"
            >
              News
            </a>
          </li>
        </ul>
        <h3>
          Ecosystem
        </h3>
        <ul>
          <li>
            <a
              href="https://router.vuejs.org"
              rel="noopener"
              target="_blank"
            >
              vue-router
            </a>
          </li>
          <li>
            <a
              href="https://vuex.vuejs.org"
              rel="noopener"
              target="_blank"
            >
              vuex
            </a>
          </li>
          <li>
            <a
              href="https://github.com/vuejs/vue-devtools#vue-devtools"
              rel="noopener"
              target="_blank"
            >
              vue-devtools
            </a>
          </li>
          <li>
            <a
              href="https://vue-loader.vuejs.org"
              rel="noopener"
              target="_blank"
            >
              vue-loader
            </a>
          </li>
          <li>
            <a
              href="https://github.com/vuejs/awesome-vue"
              rel="noopener"
              target="_blank"
            >
              awesome-vue
            </a>
          </li>
        </ul>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": undefined,
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": undefined,
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.7",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "hasOwnProperty": [Function],
    "msg": "new message",
  },
  "rootVM": Object {},
  "wrapperElement": <div
    class="hello"
  >
    <h1>
      new message
    </h1>
    <p>
       For a guide and recipes on how to configure / customize this project,
      <br />
       check out the 
      <a
        href="https://cli.vuejs.org"
        rel="noopener"
        target="_blank"
      >
        vue-cli documentation
      </a>
      . 
    </p>
    <h3>
      Installed CLI Plugins
    </h3>
    <ul>
      <li>
        <a
          href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-babel"
          rel="noopener"
          target="_blank"
        >
          babel
        </a>
      </li>
      <li>
        <a
          href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint"
          rel="noopener"
          target="_blank"
        >
          eslint
        </a>
      </li>
    </ul>
    <h3>
      Essential Links
    </h3>
    <ul>
      <li>
        <a
          href="https://vuejs.org"
          rel="noopener"
          target="_blank"
        >
          Core Docs
        </a>
      </li>
      <li>
        <a
          href="https://forum.vuejs.org"
          rel="noopener"
          target="_blank"
        >
          Forum
        </a>
      </li>
      <li>
        <a
          href="https://chat.vuejs.org"
          rel="noopener"
          target="_blank"
        >
          Community Chat
        </a>
      </li>
      <li>
        <a
          href="https://twitter.com/vuejs"
          rel="noopener"
          target="_blank"
        >
          Twitter
        </a>
      </li>
      <li>
        <a
          href="https://news.vuejs.org"
          rel="noopener"
          target="_blank"
        >
          News
        </a>
      </li>
    </ul>
    <h3>
      Ecosystem
    </h3>
    <ul>
      <li>
        <a
          href="https://router.vuejs.org"
          rel="noopener"
          target="_blank"
        >
          vue-router
        </a>
      </li>
      <li>
        <a
          href="https://vuex.vuejs.org"
          rel="noopener"
          target="_blank"
        >
          vuex
        </a>
      </li>
      <li>
        <a
          href="https://github.com/vuejs/vue-devtools#vue-devtools"
          rel="noopener"
          target="_blank"
        >
          vue-devtools
        </a>
      </li>
      <li>
        <a
          href="https://vue-loader.vuejs.org"
          rel="noopener"
          target="_blank"
        >
          vue-loader
        </a>
      </li>
      <li>
        <a
          href="https://github.com/vuejs/awesome-vue"
          rel="noopener"
          target="_blank"
        >
          awesome-vue
        </a>
      </li>
    </ul>
  </div>,
}
`;
